
#### ChangeSet ####
2009-04-09 04:13:41-04:00, stenn@whimsy.udel.edu 
  [Sec 1144] limited buffer overflow in ntpq.  CVE-2009-0159

==== ntpq/ntpq.c ====
2009-04-09 04:13:30-04:00, stenn@whimsy.udel.edu +2 -2
  [Sec 1144] limited buffer overflow in ntpq.  CVE-2009-0159

--- 1.65/ntpq/ntpq.c	2006-07-26 00:55:41 -07:00
+++ 1.66/ntpq/ntpq.c	2009-04-09 01:13:30 -07:00
@@ -3185,9 +3185,9 @@ cookedprint(
 				if (!decodeuint(value, &uval))
 				    output_raw = '?';
 				else {
-					char b[10];
+					char b[12];
 
-					(void) sprintf(b, "%03lo", uval);
+					(void) snprintf(b, sizeof b, "%03lo", uval);
 					output(fp, name, b);
 				}
 				break;
